src/test/java/uk/gov/hmcts/reform/pdf/service/endpoint/v1/DeprecatePDFGenerationEndpointTest.java

```diff
@@ -0,0 +1,37 @@
+package uk.gov.hmcts.reform.pdf.service.endpoint.v1;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.mock.web.MockHttpServletResponse;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.web.servlet.MockMvc;
+
+import java.nio.charset.Charset;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.fileUpload;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest
+@AutoConfigureMockMvc
+public class DeprecatePDFGenerationEndpointTest {
+
+    private static final String API_URL = "/api/v1/pdf-generator/html";
+
+    @Autowired
+    private MockMvc webClient;
+
+    @Test
+    public void should_return_deprecated_api() throws Exception {
```

Comment:
perhaps 'should add deprecation warning header to response'?

Reply:
Yep, that looks closer to the actual test.